### PR TITLE
Add address

### PR DESCRIPTION
**Pull request by Kacper-Berganski**

### CODE REVIEW

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.html [hunk 1]
```diff
@@ -73,20 +70,20 @@ <h1>Add Address</h1>
...
+                Street is too long !<i class="fa fa-exclamation"></i>
               </div>
             </div>
 
           <br>
 
-  <button type="submit"> Save</button>
-  <button type="button" (click)="form.reset()">Clear</button>
+  <button class="buttonS" type="submit" (click)= showSuccess()> Save <fa-icon class="icon" [icon]="faSave"></fa-icon></button>
+  <button class="buttonC" type="button" (click)="form.reset(); clean()">Clear <fa-icon class="icon" [icon]="faTimes"></fa-icon></button>
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.html:86]: Changes names due to convention e.g. "btn-save"

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+.buttonC,
+.buttonS {
+  font-size: 1.9rem;
+  position: relative;
+  background: transparent;
+  border: none;
+  left: 0.5rem;
+  padding: 0.1rem 0.5rem;
+  transition: 0.3s transform;
+}
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss:29]: If they both have the same properties why do you need two classes?

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    font-size: 2rem;
+  }
+
+  .label {
+    font-size: 1.2rem;
+  }
+  .buttonC,
+  .buttonS {
+    font-size: 1.7rem;
+  }
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss:78]: Why did ou separate these values ?

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.alert-danger {
+  margin-top: 0.5rem;
+}
+
+@media only screen and (min-width: 100px) and (max-width: 550px) {
+  .inputNgForm,
+  .selectCountry {
+    width: 16rem;
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss:63]: the width should not be in rem.
How do you imagine this size?  as 16 normal height of one letter on your site? 

Better use % for this property.

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  background: transparent;
+  border: none;
+  left: 0.5rem;
+  padding: 0.1rem 0.5rem;
+  transition: 0.3s transform;
+}
+
+.buttonC:hover {
+  transform: scale(1.2);
+  color: rgb(253, 77, 77);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss:33]: Probably you will use the same button colors in rest places on your site. Use colors.scss.

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts [hunk 6]
```diff
@@ -1,48 +1,117 @@
...
+  address: AddressForCreation = {
+    countryId: '',
+    city: '',
+    street: '',
+    zipCode: '',
+  };
+  form: MyFormGroup = null;
+  faSave = faSave;
+  faTimes = faTimes;
+  baseUrl: string = 'https://localhost:44391/api/addresses';
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts:34]: why do you need this address here? It should be used only in service.

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts [hunk 7]
```diff
@@ -1,48 +1,117 @@
...
+          : this.address.countryId;
+
+    this.form = new MyFormGroup({
+      countryId: new FormControl(initialCountry, [
+        Validators.required]),
+      zipCode: new FormControl(this.address.zipCode, [
+        Validators.required,
+        Validators.minLength(5),
+        Validators.maxLength(7),
+        Validators.pattern('[0-9]{2}\-[0-9]{3}'),
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts:62]: Magic pattern, please put it into a properly named variable first and then use it in this Validators.pattern.

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts [hunk 8]
```diff
@@ -1,48 +1,117 @@
...
+      ]),
+      city: new FormControl(this.address.city, [
+        Validators.required,
+        Validators.minLength(3),
+        Validators.maxLength(30),
+        Validators.pattern('[a-zA-Z ]*'),
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts:68]: aa

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts [hunk 9]
```diff
@@ -1,48 +1,117 @@
...
+  }
 
   ngOnInit() {
+
   }
 
-  saveAddress(){
-    console.log("first step saving", this.form.value);
+  saveAddress() {
+    console.log('first step saving', this.form.value);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts:84]: throw away console.logs after end of work.

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts [hunk 10]
```diff
@@ -1,48 +1,117 @@
...
-     this.addressService.create(addressToCreate).subscribe(response => {
-       console.log("Subscribe for creation")
-     });
-  
-  };
-}
+      street: this.form.value.street,
+    };
+
+    console.log('saving', addressToCreate);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts:93]: aa

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts [hunk 11]
```diff
@@ -1,48 +1,117 @@
...
+    this.addressService.create(addressToCreate).subscribe((response) => {
+      console.log('Subscribe for creation');
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts:95]: aa

File: CustomerManagementPanel/src/app/addresses/addresses.component.html [hunk 12]
```diff
@@ -16,25 +16,32 @@
         </tr>
         </thead>
         <tbody>
-        <tr *ngFor="let address of addressesList; index as i">
+        <tr *ngFor="let address of orderedAddresses; index as i">
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses.component.html:19]: After paging you have property address.orderedNumber, you dont need index anymore.

File: CustomerManagementPanel/src/app/addresses/addresses.component.ts [hunk 13]
```diff
@@ -24,15 +26,14 @@ export class AddressesComponent implements OnInit {
       private route: ActivatedRoute,
       private addressService: AddressesService
     ) { 
-      this.route.data.subscribe(value => {      
-        this.addressesList = value["addressesList"];
-        console.log(this.addressesList);
-      });
+     console.log('constructor');
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses.component.ts:29]: aa

File: CustomerManagementPanel/src/app/addresses/addresses.component.ts [hunk 14]
```diff
@@ -42,7 +43,10 @@ export class AddressesComponent implements OnInit {
     }
 
     addressDetails(addressToDelete:Address){
-
     }
   
+    onPageChanged(event: MyPager) {
+      console.log("łapiemy event", event);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses.component.ts:49]: aa

File: CustomerManagementPanel/src/app/addresses/addresses.component.ts [hunk 15]
```diff
@@ -24,15 +26,14 @@ export class AddressesComponent implements OnInit {
...
-        this.addressesList = value["addressesList"];
-        console.log(this.addressesList);
-      });
+     console.log('constructor');
     }
   
     ngOnInit(): void {
     }
 
+
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses.component.ts:35]: remove one redundant line

File: CustomerManagementPanel/src/app/addresses/addresses.service.ts [hunk 16]
```diff
@@ -48,6 +48,16 @@ constructor(private httpClient:HttpClient) {
         );
     }
 
+    getAllCountries() : Observable<Country[]>{
+        return this.httpClient.get<Country[]>(this.baseUrl + "/GetAllCountries").pipe(
+          tap(response =>
+              {
+                  console.log("From services:", response);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses.service.ts:55]: aa

File: CustomerManagementPanel/src/app/app.module.ts [hunk 17]
```diff
@@ -40,6 +39,7 @@ import { PaginationComponent } from './shared/pagination/pagination.component';
     InProgressComponent,
     LoginPanelComponent,
     SignUpPanelComponent,
+    // PaginationComponent
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/app.module.ts:42]: ??

File: CustomerManagementPanel/src/app/clients/client-list/client-list.component.html [hunk 18]
```diff
@@ -16,23 +16,29 @@
     </tr>
     </thead>
     <tbody>
-    <tr *ngFor="let customer of customerObjectList; index as i">
+    <tr *ngFor="let customer of orderedCustomers; index as i">
         <th scope="row">{{ i + 1 }}</th>
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/clients/client-list/client-list.component.html:20]: aa

File: CustomerManagementPanel/src/app/clients/client-list/client-list.component.ts [hunk 19]
```diff
@@ -17,7 +18,8 @@ export class ClientListComponent implements OnInit {
   isAddingMode: boolean = false;
   newCustomer: Customer = null;
   id: number = 11114;
-  customerObjectList: Customer[];
+  customerObjectList: Customer[]; // dostajemy z resolvera (czyli z servera liste uzytkownikow)
+  orderedCustomers: OrderedItem[]; // wpada z pagination, juz w odpowiedniej ilosci item=customer, OrderNumber
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/clients/client-list/client-list.component.ts:22]: Remove comments from me

File: CustomerManagementPanel/src/app/clients/client-list/client-list.component.ts [hunk 20]
```diff
@@ -65,13 +66,10 @@ export class ClientListComponent implements OnInit {
     newCustomer.id = (++this.id).toString();
     // CUSTOMERLIST.push(newCustomer);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/clients/client-list/client-list.component.ts:67]: ??

File: CustomerManagementPanel/src/app/clients/client-list/client-list.component.ts [hunk 21]
```diff
@@ -65,13 +66,10 @@ export class ClientListComponent implements OnInit {
...
-}
-
-
-
-
-
-
 
+  onPageChanged(event: MyPager) {
+    console.log("łapiemy event", event);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/clients/client-list/client-list.component.ts:72]: aa

File: CustomerManagementPanel/src/app/clients/edit-customer/edit-customer.component.scss [hunk 22]
```diff
@@ -1,82 +1,61 @@
...
+.buttonC, .saveNCustomer{
+    font-size: 1.9rem;
+    position: relative;
+    background: transparent;
+    border: none;
+    left: .5rem;
+    cursor: pointer;
+    padding: .1rem .5rem ;
+    transition: .3s transform;
+    color: rgb(158, 158, 158);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/clients/edit-customer/edit-customer.component.scss:29]: aa

File: CustomerManagementPanel/src/app/clients/edit-customer/edit-customer.component.scss [hunk 23]
```diff
@@ -1,82 +1,61 @@
...
 }
 
-.classCss-errors {
-    color: red;
+.buttonC:hover{
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/clients/edit-customer/edit-customer.component.scss:32]: aa

File: CustomerManagementPanel/src/app/clients/edit-customer/edit-customer.component.ts [hunk 24]
```diff
@@ -39,14 +43,21 @@ export class EditCustomerComponent implements OnInit {
       }
   });
 
+  let country = this.copyOfCustomer.address?.country;
+
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/clients/edit-customer/edit-customer.component.ts:47]: remove redundant lines

**Review by AleksandraX — CHANGES_REQUESTED** [review]

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss (hunk 3 above)
Comment (Kacper-Berganski) [reply CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss:78]: Change the size of the displayed website of the page.

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+  background: transparent;
+  border: none;
+  left: 0.5rem;
+  padding: 0.1rem 0.5rem;
+  transition: 0.3s transform;
+}
+
+.bnt-clear:hover {
+  transform: scale(1.2);
+  color: var(--colorClear);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss:33]: "colorAlmostRed"

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.bnt-save:hover {
+  transform: scale(1.2);
+  color: rgb(0, 116, 0);
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss:38]: color variable

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.inputNgForm,
```
Comment (AleksandraX) [inline CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.scss:41]: avoid such a classes naming

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts (hunk 7 above)
Comment (AleksandraX) [reply CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts:62]: let zipCodePattern = '[0-9]{2}\-[0-9]{3}' // XX-XXX
let mailPattern = "....." // xxxxx@xx.xx

File: CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts (hunk 8 above)
Comment (AleksandraX) [reply CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts:68]: aa

**Review by AleksandraX — CHANGES_REQUESTED** [review]

**Pull request merged**